### ISSUE TITLE
Optional File Input paths not being resolved to local path in JES backend - Cromwell 24

### ISSUE THREAD

**Issue by tlangs** [broadinstitute/cromwell#1891]
When an optional file input is given to a task, the file path is not resolved to the local path in the JES backend. Non-optional Files are resolved appropriately. 

Example failing task:
```
task ValidateSamFile {
  File input_bam
  File? ref_fasta

  command {
    java -Xmx4000m -jar /usr/gitc/picard.jar \
      BlahBlahThisIsACommand \
      INPUT=${input_bam} \
      ${"REFERENCE_SEQUENCE=" + ref_fasta}
  }
  runtime {
    docker: "broadinstitute/genomes-in-the-cloud:2.2.4-1469632282"
    memory: "7 GB"
    disks: "local-disk " + disk_size + " HDD"
    preemptible: preemptible_tries
  }
  output {
    File report = "${report_filename}"
  }
}
```

Output from stderr:
```
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/cromwell_root/tmp
[Mon Jan 23 17:32:28 UTC 2017] picard.sam. BlahBlahThisIsACommand INPUT=/cromwell_root/broad-dsde-methods/cromwell-execution-24/PairedEndSingleSampleWorkflow/5802ded7-ba93-4186-b680-5a7cb2c7f62c/call-SortAndFixReadGroupBam/shard-1/H06HDADXX130110.2.ATCACGAT.20k_reads.sorted.bam \
REFERENCE_SEQUENCE=gs:/broad-references/hg38/v0/Homo_sapiens_assembly38.fasta    
[Mon Jan 23 17:32:28 UTC 2017] Executing as root@4543b0496fa1 on Linux 3.16.0-0.bpo.4-amd64 amd64; OpenJDK 64-Bit Server VM 1.8.0_91-8u91-b14-1~bpo8+1-b14; Picard version: 2.5.0-gradle-SNAPSHOT
[Mon Jan 23 17:32:28 UTC 2017] picard.sam.ValidateSamFile done. Elapsed time: 0.00 minutes.
Runtime.totalMemory()=119013376
To get help, see http://broadinstitute.github.io/picard/index.html#GettingHelp
Exception in thread "main" htsjdk.samtools.SAMException: Cannot read non-existent file: /cromwell_root/gs:/broad-references/hg38/v0/Homo_sapiens_assembly38.fasta
	at htsjdk.samtools.util.IOUtil.assertFileIsReadable(IOUtil.java:338)
	at htsjdk.samtools.util.IOUtil.assertFileIsReadable(IOUtil.java:325)
	at picard.sam.ValidateSamFile.doWork(ValidateSamFile.java:141)
	at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:208)
	at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:95)
	at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:105)
```

**Comment by ruchim** [comment]
@cjllanwarne  is this a duplicate of #1940 ?

**Comment by tlangs** [comment]
yes

**Comment by kcibul** [comment]
Thanks @tlangs -- this is being closed as a dupe of #1940 which is being worked.  I'll add you as a watcher there.
